Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -145,8 +164,47 @@ public struct HTTPResponseHead: Equatable {
     }
 }
 
-fileprivate typealias HTTPHeadersStorage = [String: [(String, String)]] // [lowerCasedName: [(originalCaseName, value)]
+/// The Index for a header name or value that points into the underlying `ByteBuffer`.
+struct HTTPHeaderIndex {
+    let start: Int
+    let length: Int
+}
+
+/// Struct which holds name, value pairs.
+struct HTTPHeader {
+    let name: HTTPHeaderIndex
+    let value: HTTPHeaderIndex
+}
+
+private extension ByteBuffer {
+    func equalCaseInsensitiveASCII(view: String.UTF8View, at index: HTTPHeaderIndex) -> Bool {
+        guard view.count == index.length else {
+            return false
+        }
+        return withVeryUnsafeBytes { buffer in
+            // This should never happens as we control when this is called. Adding and assert to ensure this.
+            assert(index.start <= self.capacity - index.length)
+            let address = buffer.baseAddress!.assumingMemoryBound(to: UInt8.self)
+            var idx = 0
+            for byte in view {
+                guard address.advanced(by: index.start + idx).pointee & 0xdf == byte & 0xdf else {
```

Comment:
Just to clarify, did you decide not to have this enforce that the bytes actually are ASCII?